### DISCUSSION TITLE
Azure AD auth not working outside of organization

### DISCUSSION THREAD

**Discussion by coumbsek** [requarks/wiki#5388] [Help / Questions]
Hi,
I'm trying to setup a Azure AD authentication. I have created my app and set it to accep Multitenant and personal accounts
If I use as OpenID Metadata endpoint `https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration` as Microsoft is saying I have the
> Oops, something went wrong...
> Invalide email / username or password.

message as soon as I clic on Azure AD login option, I can not event enter username/password values.
If I use the address found in the wiki.js doc `https://login.microsoftonline.com/YOUR_TENANT_ID/v2.0/.well-known/openid-configuration` replacing `YOUR_TENANT_ID` by my tenant id I can connect to account inside my organization but not ouside (which is logical from a Microsoft point of view).
Can anyone help me ?

**Comment by amal-viswa** [comment]
Hey, how did you setup the ad authentication  ? , i'm facing an issue with ad authentication part i've followed the wiki.js official documentation for azure ad official , but still the same error which you're facing . can you help me resolve it 

error:

![Screenshot 2022-08-23 at 18-12-26 Error](https://user-images.githubusercontent.com/90927394/186192790-8236a080-025f-4986-a1c9-bd1717380263.png)

**Reply by coumbsek** [reply]
Hi, we did not solve it unfortunately !
We are waiting for an update / MaJ that would apport this bug correction or a documentation clarification (but I have doubts that this is just an issue about configuration)

**Reply by amal-viswa** [reply]
hi, 
i'm having issue for the single tenant configuration , not even working the for internal organization. so in your case its working right  ?  . have you done any another steps additionally

**Reply by coumbsek** [reply]
We have it working for our organization indeed.
I do not remember doing something particular on the wiki js side but the azure part was quite a nightmare from my point of view.

I also found back the post I was looking for on the slack, the callback URL in the  dock is not correct. Take the one that appears on the wiki.js azure ad authentication configuration page. It should look like : https://your.wiki.domain.com/login/UUID/callback
Check that this is correct in Azure AD config page, if you strictly follow the doc it is certainly the error you are looking for.
